lib/publiccloud/instance.pm

```diff
@@ -427,43 +429,45 @@ sub wait_for_ssh {
         $sshout .= "as expected by stopping: OK.\n" if $args{wait_stop};
     }    # endif
 
-    # check also remote system is up and running:
-    #   SSH host key is not checked and master socket is not used
+    # Check also remote system is up and running:
     my $retry = 0;    # count retries of unexpected sysout
-    if ($args{systemup_check} and isok($exit_code)) {
-        my $ssh_opts = $self->ssh_opts() . ' -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ControlPath=none';
-        while (($duration = time() - $start_time) < $args{timeout}) {
-            # timeout recalculated removing consumed time until now
-            # We don't support password authentication so it would just block the terminal
-            $sysout = $self->ssh_script_output(cmd => 'sudo systemctl is-system-running', ssh_opts => $ssh_opts,
-                timeout => $args{timeout} - $duration, proceed_on_failure => 1, username => $args{username});
-            # result check
-            if ($sysout =~ m/initializing|starting/) {    # still starting
-                $exit_code = undef;
-            }
-            elsif ($sysout =~ m/running/) {    # startup OK
-                $exit_code = 0;
-                $sysout .= "\nSystem successfully booted";
-                last;
-            }
-            elsif ($sysout =~ m/degraded/) {    # up but with failed services to collect
-                $exit_code = 0;
-                $sysout .= "\nSystem booted, but some services failed:\n" .
-                  $self->ssh_script_output(cmd => 'sudo systemctl --failed', ssh_opts => $ssh_opts,
-                    proceed_on_failure => 1, username => $args{username});
-                last;
-            }
-            elsif ($sysout =~ m/maintenance|stopping|offline|unknown/) {
-                $exit_code = 1;
-                $sysout .= "\nCan not reach systemd target";
-                last;
-            }
-            else {    # other outcome or connection refused: retry/reloop
-                $exit_code = 2;
-                ++$retry;
-            }    # endif
-            sleep $delay;
-        }    # end loop
+    if (isok($exit_code)) {
+        if ($args{systemup_check}) {
```

Comment:
!! I think this changes the behaviour. In the old code, line 468 is only executed if `$args{systemup_check} and isok($exit_code)` was true, here it is executed if only `isok($exit_code)` is true.

Is that really what we want? :thinking:

Reply:
and why do you think we need to do ssh-keyscan in calls w/o `systemup_check` ?